src/test/rustdoc/issue-82209.rs

```diff
@@ -0,0 +1,8 @@
+#![deny(broken_intra_doc_links)]
+pub enum Foo {
+    Bar {
+        abc: i32,
+        /// [Self::Bar::abc]
```

Comment:
Can you also test that this generates the right link? There's instructions at https://rustc-dev-guide.rust-lang.org/rustdoc-internals.html#dotting-is-and-crossing-ts and examples in the other files in this directory.

Please also move this to `rustdoc/intra-doc`, with the rest of the intra-doc link tests.

Reply:
I've added a test for the generation of the link. I used a `#![crate_name = "foo"]`at the beginning since I couldn't get it to work with `issue-82209` as the name of the file. Is the repetition of "foo" okay?

Reply:
Yes, that's fine. Fyi the way to do it with dashes is to change them to underscores in the filename,  but this works too.

Reply:
Thanks for the explanation.